libraries/Parsers/Microsoft.Bot.Builder.Parsers.LU/Parser/LUResource.cs

```diff
@@ -0,0 +1,407 @@
+            },
+        };
+    }
+
+    internal class SectionTypeConverter : JsonConverter
+    {
+        public static readonly SectionTypeConverter Singleton = new SectionTypeConverter();
+
+        public override bool CanConvert(Type t) => t == typeof(SectionType) || t == typeof(SectionType?);
+
+        public override object ReadJson(JsonReader reader, Type t, object existingValue, JsonSerializer serializer)
+        {
+            if (reader.TokenType == JsonToken.Null)
+            {
+                return null;
+            }
+
+            var value = serializer.Deserialize<string>(reader);
+            switch (value)
+            {
+                case "entitySection":
+                    return SectionType.EntitySection;
+                case "newEntitySection":
+                    return SectionType.NewEntitySection;
+                case "simpleIntentSection":
+                    return SectionType.SimpleIntentSection;
+                case "nestedIntentSection":
+                    return SectionType.NestedIntentSection;
+                case "importSection":
+                    return SectionType.ImportSection;
+                case "modelInfoSection":
+                    return SectionType.ModelInfoSection;
+                case "qnaSection":
+                    return SectionType.QnaSection;
+            }
+
+            throw new Exception("Cannot unmarshal type SectionType");
+        }
+
+        public override void WriteJson(JsonWriter writer, object untypedValue, JsonSerializer serializer)
+        {
+            if (untypedValue == null)
+            {
+                serializer.Serialize(writer, null);
+                return;
+            }
+
+            var value = (SectionType)untypedValue;
+            switch (value)
+            {
+                case SectionType.EntitySection:
+                    serializer.Serialize(writer, "entitySection");
+                    return;
+                case SectionType.NewEntitySection:
+                    serializer.Serialize(writer, "newEntitySection");
+                    return;
+                case SectionType.SimpleIntentSection:
+                    serializer.Serialize(writer, "simpleIntentSection");
+                    return;
+                case SectionType.NestedIntentSection:
+                    serializer.Serialize(writer, "nestedIntentSection");
+                    return;
+                case SectionType.ImportSection:
+                    serializer.Serialize(writer, "importSection");
+                    return;
+                case SectionType.ModelInfoSection:
+                    serializer.Serialize(writer, "modelInfoSection");
+                    return;
+                case SectionType.QnaSection:
+                    serializer.Serialize(writer, "qnaSection");
+                    return;
+            }
+
+            throw new Exception("Cannot marshal type EntitySectionType");
```

Comment:
Same feedbqack around updating the error message to include the type name that failed.